Sources/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Macros must be built for the build machine, not the host.
+include(ExternalProject)
+ExternalProject_Add(TestingMacros
+  CMAKE_CACHE_DEFAULT_ARGS "-DCMAKE_MODULE_PATH:FILEPATH=${CMAKE_MODULE_PATH}"
+  PREFIX "tm"
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/TestingMacros"
+  INSTALL_COMMAND "")
+ExternalProject_Get_Property(TestingMacros BINARY_DIR)
+if(CMAKE_HOST_WIN32)
+  set(TestingMacrosPath "${BINARY_DIR}/TestingMacros.exe#TestingMacros")
+else()
+  set(TestingMacrosPath "${BINARY_DIR}/TestingMacros#TestingMacros")
```

Comment:
For including in the toolchain we'll want to build these as a library (the driver adds a `-load-plugin` for `bin` on Windows and `lib/swift/host/plugins` on other platforms).